From 9a17140485c6749f6ceb70e551813aeff8595aa1 Mon Sep 17 00:00:00 2001
From: John Morris <john@zultron.com>
Date: Mon, 6 Oct 2014 14:41:01 -0500
Subject: [PATCH 2/2] include/asm-x86_64/GNUmakefile.am doesn't honor
 $(includedir)

`./configure --includedir=...` isn't honored by
`include/asm-x86_64/GNUmakefile.am`, since it redefines `$(includedir)`
within the file in order to install into a sub-directory.

Fix by using another variable prefix.

Signed-off-by: John Morris <john@zultron.com>
---
 base/include/asm-arm/GNUmakefile.am       | 18 +++++++++---------
 base/include/asm-i386/GNUmakefile.am      | 10 +++++-----
 base/include/asm-m68k/GNUmakefile.am      | 10 +++++-----
 base/include/asm-m68knommu/GNUmakefile.am | 10 +++++-----
 base/include/asm-ppc/GNUmakefile.am       | 10 +++++-----
 base/include/asm-x86/GNUmakefile.am       | 14 +++++++-------
 base/include/asm-x86_64/GNUmakefile.am    | 10 +++++-----
 7 files changed, 41 insertions(+), 41 deletions(-)

diff --git a/base/include/asm-arm/GNUmakefile.am b/base/include/asm-arm/GNUmakefile.am
index b53f909..2dae08d 100644
--- a/base/include/asm-arm/GNUmakefile.am
+++ b/base/include/asm-arm/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = $(prefix)/include/asm-arm
+includeasmdir = $(includedir)/asm-arm
 
-nobase_include_HEADERS = \
+includeasm_HEADERS = \
 		rtai_atomic.h \
 		rtai_fpu.h \
 		rtai.h \
@@ -13,33 +13,33 @@ nobase_include_HEADERS = \
 		rtai_usi.h \
 		rtai_vectors.h
 if CONFIG_ARCH_PXA
-nobase_include_HEADERS += \
+nobase_includeasm_HEADERS = \
 		arch-pxa/rtai_arch.h \
 		arch-pxa/rtai_exports.h \
 		arch-pxa/rtai_timer.h
 endif
 if CONFIG_ARCH_IMX
-nobase_include_HEADERS += \
+nobase_includeasm_HEADERS = \
 		arch-imx/rtai_arch.h \
 		arch-imx/rtai_exports.h \
 		arch-imx/rtai_timer.h
 endif
 if CONFIG_ARCH_EP9301
-nobase_include_HEADERS += \
+nobase_includeasm_HEADERS = \
 		arch-ep9301/rtai_arch.h \
 		arch-ep9301/rtai_exports.h \
 		arch-ep9301/rtai_timer.h
 endif
 if CONFIG_ARCH_AT91
-nobase_include_HEADERS += \
+nobase_includeasm_HEADERS = \
 		arch-at91/rtai_arch.h \
 		arch-at91/rtai_exports.h \
 		arch-at91/rtai_timer.h
 endif
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	$(LN_S) -f asm-arm $(DESTDIR)$(includedir)/../asm
-	$(LN_S) -f arch-@RTAI_TARGET_SUBARCH@ $(DESTDIR)$(includedir)/arch
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	$(LN_S) -f asm-arm $(DESTDIR)$(includedir)/asm
+	$(LN_S) -f arch-@RTAI_TARGET_SUBARCH@ $(DESTDIR)$(includeasmdir)/arch
 
 EXTRA_DIST = rtai_hal.immed rtai_hal.piped
diff --git a/base/include/asm-i386/GNUmakefile.am b/base/include/asm-i386/GNUmakefile.am
index b167ba1..e0adf56 100644
--- a/base/include/asm-i386/GNUmakefile.am
+++ b/base/include/asm-i386/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = @includedir@/asm-i386
+includeasmdir = $(includedir)/asm-i386
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -16,8 +16,8 @@ include_HEADERS = \
 		rtai_vectors.h
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-i386 $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-i386 $(DESTDIR)$(includedir)/asm
 
 EXTRA_DIST = rtai_hal.piped rtai_hal.immed
diff --git a/base/include/asm-m68k/GNUmakefile.am b/base/include/asm-m68k/GNUmakefile.am
index 1f44344..df86d8b 100644
--- a/base/include/asm-m68k/GNUmakefile.am
+++ b/base/include/asm-m68k/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = @includedir@/asm-m68k
+includeasmdir = $(includedir)/asm-m68k
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -15,6 +15,6 @@ include_HEADERS = \
 		rtai_vectors.h
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-m68k $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-m68k $(DESTDIR)$(includedir)/asm
diff --git a/base/include/asm-m68knommu/GNUmakefile.am b/base/include/asm-m68knommu/GNUmakefile.am
index 1af9412..6bc53f2 100644
--- a/base/include/asm-m68knommu/GNUmakefile.am
+++ b/base/include/asm-m68knommu/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = @includedir@/asm-m68knommu
+includeasmdir = $(includedir)/asm-m68knommu
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -15,6 +15,6 @@ include_HEADERS = \
 		rtai_vectors.h
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-m68knommu $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-m68knommu $(DESTDIR)$(includedir)/asm
diff --git a/base/include/asm-ppc/GNUmakefile.am b/base/include/asm-ppc/GNUmakefile.am
index 21afb89..dfd877e 100644
--- a/base/include/asm-ppc/GNUmakefile.am
+++ b/base/include/asm-ppc/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = $(prefix)/include/asm-ppc
+includeasmdir = $(includedir)/asm-ppc
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -15,6 +15,6 @@ include_HEADERS = \
 		rtai_vectors.h
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-ppc $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-ppc $(DESTDIR)$(includedir)/asm
diff --git a/base/include/asm-x86/GNUmakefile.am b/base/include/asm-x86/GNUmakefile.am
index 9024af1..8b974e0 100644
--- a/base/include/asm-x86/GNUmakefile.am
+++ b/base/include/asm-x86/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = @includedir@/asm-x86
+includeasmdir = $(includedir)/asm-x86
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -16,14 +16,14 @@ include_HEADERS = \
 		rtai_vectors.h
 
 if CONFIG_I386
-include_HEADERS += \
+includeasm_HEADERS += \
 	rtai_atomic_32.h \
 	rtai_fpu_32.h \
 	rtai_hal_32.h \
 	rtai_sched_32.h \
 	rtai_lxrt_32.h
 else
-include_HEADERS += \
+includeasm_HEADERS += \
 	rtai_atomic_64.h \
 	rtai_fpu_64.h \
 	rtai_hal_64.h \
@@ -32,8 +32,8 @@ include_HEADERS += \
 endif
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-x86 $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-x86 $(DESTDIR)$(includedir)/asm
 
 #EXTRA_DIST = rtai_hal.piped rtai_hal.immed
diff --git a/base/include/asm-x86_64/GNUmakefile.am b/base/include/asm-x86_64/GNUmakefile.am
index e84ab72..486d689 100644
--- a/base/include/asm-x86_64/GNUmakefile.am
+++ b/base/include/asm-x86_64/GNUmakefile.am
@@ -1,6 +1,6 @@
-includedir = $(prefix)/include/asm-x86_64
+includeasmdir = $(includedir)/asm-x86_64
 
-include_HEADERS = \
+includeasm_HEADERS = \
 		rtai.h \
 		rtai_atomic.h \
 		rtai_fpu.h \
@@ -16,8 +16,8 @@ include_HEADERS = \
 		rtai_vectors.h
 
 install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(includedir)
-	rm -f $(DESTDIR)$(includedir)/../asm
-	$(LN_S) asm-x86_64 $(DESTDIR)$(includedir)/../asm
+	$(mkinstalldirs) $(DESTDIR)$(includeasmdir)
+	rm -f $(DESTDIR)$(includedir)/asm
+	$(LN_S) asm-x86_64 $(DESTDIR)$(includedir)/asm
 
 EXTRA_DIST = rtai_hal.piped rtai_hal.immed
-- 
1.9.1

